src/application/engine/game_object/GameObject.java

```diff
@@ -0,0 +1,242 @@
+package application.engine.game_object;
+
+import java.util.Observable;
+import java.util.concurrent.ConcurrentMap;
+
+import org.datacontract.schemas._2004._07.Ball_of_Duty_Server_DTO.GameObjectDTO;
+
+import application.engine.game_object.physics.Physics;
+import application.util.Vector2;
+import javafx.geometry.Point2D;
+import javafx.scene.canvas.GraphicsContext;
+import javafx.scene.image.Image;
+
+/**
+ * A game object is all objects that can be in a game. A ClientMap usually handles game objects without knowing who might be extending the
+ * game object.
+ * 
+ * @author Gruppe6
+ *
+ */
+public class GameObject extends Observable
+{
+
+    protected Body body;
+    protected Physics physics;
+    protected View view;
+    protected Health health;
+    protected Weapon weapon;
+    private int id;
+    private boolean destroyed;
+
+  
+
+    /**
+     * Creates a copy of a game object based on another game object.
+     *
+     * @param gameObject
+     *            The game object to be copied.
+     */
+    public GameObject(GameObject gameObject)
```

Comment:
http://stackoverflow.com/questions/869033/how-do-i-copy-an-object-in-java
also, when do we use this constructor?